tf2_ros/src/tf2_ros/buffer.py

```diff
@@ -275,3 +279,6 @@ def _on_new_data():
         fut.add_done_callback(lambda _: self._remove_callback(_on_new_data))
 
         return fut
+
+    def wait_for_transform(target_frame, source_frame, time, callback):
+        return _wait_for_transform_async(target_frame, source_frame, time, callback)
```

Comment:
Instead of adding `wait_for_transform(..., callback)`, with ros2/rclpy#460 merged I think the utility #194 is asking for can be had by making the result on the future returned by `wait_for_transform_async()` contain the transform data. The interface someone would use is `fut = wait_for_transform_async(...); fut.add_done_callback(my_tf_callback)`.